usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh

```diff
@@ -61,23 +72,34 @@ for vbs in 8192 16384 32768 65536 131072; do
 		log_must $ZFS create -s -b $vbs -V $volsize $vol
 		log_must $SWAP -a $swapname
 
-		if ((mem <= min)); then		# volsize should be 2G
-			new_volsize=$(get_prop volsize $vol)
-			((new_volsize == min)) || log_fail \
-			    "Unexpected volsize: $new_volsize"
-		elif ((mem >= max)); then	# volsize should be 16G
-			new_volsize=$(get_prop volsize $vol)
-			((new_volsize == max)) || log_fail \
-			    "Unexpected volsize: $new_volsize"
-		else				# volsize should be 'mem'
-			new_volsize=$(get_prop volsize $vol)
-			((new_volsize == mem)) || log_fail \
-			    "Unexpected volsize: $new_volsize"
-		fi
+                if ((mem <= min)); then         # volsize should be 2G
```

Comment:
your proposed/original function trace with 'set -x'
https://paste.ec/paste/lt2fI+gb#hlrErHUNdf8h7G1NtkhRTuiLyD+oNR5WtsNqc6zGJm-

as you can see - we are FAILed with logic.

it is trace with 'set -x' of my variant where test is pass:
https://paste.ec/paste/HKpb9c0a#ojKms-GJmqxqfnWSTmSyuIdtuYIuxNBHlOJyIh7Vq4l